cloudinit/ssh_util.py

```diff
@@ -240,25 +240,47 @@ def render_authorizedkeysfile_paths(value, homedir, username):
         value = "%h/.ssh/authorized_keys"
     paths = value.split()
     rendered = []
+    user_authorized = default_auth
     for path in paths:
-        for macro, field in macros:
-            path = path.replace(macro, field)
+        for i in range(len(macros)):
+            macro = macros[i][0]
+            field = macros[i][1]
+            path_replaced = path.replace(macro, field)
+            # if %h or %u in path, it means it is a local key
+            if i < 2 and macro in path:
+                user_authorized = path_replaced
+            path = path_replaced
         if not path.startswith("/"):
             path = os.path.join(homedir, path)
         rendered.append(path)
-    return rendered
+    return rendered, user_authorized
 
 
 def extract_authorized_keys(username, sshd_cfg_file=DEF_SSHD_CFG):
     (ssh_dir, pw_ent) = users_ssh_info(username)
     default_authorizedkeys_file = os.path.join(ssh_dir, 'authorized_keys')
+    user_authorizedkey = default_authorizedkeys_file
+    pw_len = len(pw_ent.pw_dir)
```

Comment:
I understand you extended the use of `pw_*` variables when using `pw_len`, and I also understand this name comes from getting information from `passwd` (right?) so this is a question to the overall usage of the variables `pw_*` on this file: Is it just me or someone also thinks this is a little confusing to read? Sometimes I read it like "password" related information but then I remember it's user's information grabbed from `passwd`. Not now of course but possibly a rework in the future?

Reply:
I will use directly `len(pw_ent.pw_dir)` instead of the variable, thank you.